openfisca_uk/variables/hmrc/tax_credits/working_tax_credit/entitlement.py

```diff
@@ -46,21 +43,43 @@ class wtc_pre_minimum(Variable):
     unit = "currency-GBP"
     reference = "https://www.legislation.gov.uk/uksi/2002/2008/regulation/7"
 
-    def formula(benunit, period):
+    def formula(benunit, period, parameters):
         maximum_rate = benunit("maximum_wtc", period)
         reduction = benunit("wtc_reduction", period)
         return maximum_rate - reduction
 
 
-class working_tax_credit(Variable):
+class wtc_pre_takeup(Variable):
     label = "Working Tax Credit"
+    documentation = "Working Tax Credit entitlement, before take-up imputations."
     entity = BenUnit
     definition_period = YEAR
     value_type = float
     unit = "currency-GBP"
     reference = "https://www.legislation.gov.uk/uksi/2002/2008/regulation/7"
 
-    def formula(benunit, period):
+    def formula(benunit, period, parameters):
         maximum_rate = benunit("wtc_pre_minimum", period)
         below_minimum = benunit("tax_credits_below_minimum", period)
-        return maximum_rate * ~below_minimum * benunit("claims_wtc", period)
+        return maximum_rate * ~below_minimum
+
+
+class working_tax_credit(Variable):
+    label = "Working Tax Credit"
+    entity = BenUnit
+    definition_period = YEAR
+    value_type = float
+    unit = "currency-GBP"
+    reference = "https://www.legislation.gov.uk/uksi/2002/2008/regulation/7"
+
+    def formula(benunit, period, parameters):
+        return benunit("wtc_pre_takeup", period) * benunit("claims_wtc", period)
+
+    def formula_2021(benunit, period, parameters):
+        covid_payment = parameters(period).hmrc.tax_credits.working_tax_credit.coronavirus_payment
```

Comment:
Could we make this a single formula by setting the parameter to zero before/after? Seems cleaner to avoid separate formulas unless we need to.